### PR TITLE
[chore] Optimize get online features

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR optimizes the runtime to get online features. A feature list with 306 features is used to benchmark the runtime performance.

Current main branch takes about 14s to retrieves all online features.
original

This branch takes about 6-7s to retrieves all online features.
new

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/service/feature_materialize.py [hunk 1]
```diff
@@ -537,7 +538,8 @@ async def _get_feast_feature_store(self) -> Optional[FeastFeatureStore]:
         feast_registry = await self.feast_registry_service.get_feast_registry_for_catalog()
         if feast_registry is None:
             return None
-        return await self.feast_feature_store_service.get_feast_feature_store(feast_registry.id)
+        assert isinstance(feast_registry, FeastRegistryModel)
+        return await self.feast_feature_store_service.get_feast_feature_store(feast_registry)
```
Comment (gee-senbong) [inline featurebyte/service/feature_materialize.py:542]: By making this changes, get online features' runtime is reduced by 1s for a feature list with 306 features (14s -> 13s).

File: featurebyte/feast/patch.py [hunk 2]
```diff
@@ -126,25 +158,20 @@ def get_transformed_features_df(
...
-                columns_to_cleanup.append(feature.name)
+                df_with_features.add_column_alias(full_feature_ref, feature.name)
             elif feature.name in df_with_features.keys():
                 # Make sure the full feature name is always present
-                df_dict[full_feature_ref] = df_with_features[feature.name]
-                columns_to_cleanup.append(full_feature_ref)
+                df_with_features.add_column_alias(feature.name, full_feature_ref)
 
     # Compute transformed values and apply to each result row
-    df_with_features = pd.concat([df_with_features, pd.DataFrame(df_dict)], axis=1)
```
Comment (gee-senbong) [inline featurebyte/feast/patch.py:147]: This statement takes about 30-40% runtime of the get online features for a feature list with 306 features.

File: featurebyte/feast/patch.py [hunk 3]
```diff
@@ -158,6 +185,4 @@ def get_transformed_features_df(
             # Long name must be in dataframe.
             rename_columns[long_name] = short_name
 
-    # Cleanup extra columns used for transformation
-    df_with_features.drop(columns=columns_to_cleanup, inplace=True)
```
Comment (gee-senbong) [inline featurebyte/feast/patch.py:162]: This statements takes about 5-10% runtime of the  get online features for a feature list with 306 features.

File: featurebyte/feast/patch.py [hunk 4]
```diff
@@ -101,16 +101,48 @@ def augment_response_with_on_demand_transforms(
...
+        self._alias[alias] = column_name
+
+    def __getitem__(self, key: Any) -> Union[pd.Series, pd.DataFrame]:
+        if not isinstance(key, str) and isinstance(key, Iterable):
+            return pd.DataFrame({_key: self.__getitem__(_key) for _key in key})
+
+        if isinstance(key, str) and key in self._alias:
+            key = self._alias[key]
+
+        return super().__getitem__(key)
```
Comment (gee-senbong) [inline featurebyte/feast/patch.py:134]: By extending pandas DataFrame with alias column support, we can get rid of the expensive dataframe expansion & column dropping operations.

**Review by YS-L — APPROVED** [review]

**Pull request merged**